src/preload/preload_interface.h

```diff
@@ -434,6 +436,24 @@ struct preload_thread_locals {
   PTR(struct msghdr) notify_control_msg;
 };
 
+// The set of flags that can be set for each fd in syscallbuf_fds_disabled.
+enum syscallbuf_fd_classes {
+  // fd is invalid, all syscalls will error (syscallbuf internal use only)
+  FD_CLASS_INVALID = -1,
+  // The fd is allowed to be completely untraced. No notification to the
+  // syscall buf is required.
+  FD_CLASS_UNTRACED = 0x0,
+  // This is the most conservative option. All operations on this fd are
+  // always traced. If there is a conflict between other options, this one
+  // should be chosen.
+  FD_CLASS_TRACED   = 0x1,
+  // This fd either refers to a /proc/<pid>/mem or is untrace (if this as
+  // is shared with another fd table)
```

Comment:
If more than one fd table is using this address space, we shouldn't have any entries set to FD_CLASS_PROC_MEM, should we?

Reply:
This is now handled with the fdt_uniform flag. Otherwise, we'd have to go and flips the flags back and forth every time somebody vforks or that vfork execs or exits, which seemed suboptimal, especially, since the replay assist part of it still can work fine (only the non-blocking part if problematic).

Reply:
Is flipping the flags really that bad? It seems to me it would be conceptually simpler.

Reply:
Probably not. I suppose you're unlikely to have many proc/mem/ files open making it about the same amount of memory traffic. I can switch it to do that. That said, I may not get to it for a few days, since I have some other high priority items pending.

Reply:
I tried doing it the other way, and it just looked more complex to me. You basically need to keep this state anyway and this way we, we get to be more precise, about only turning off the part we need. However, if it does turn out we want the other thing, there shouldn't be any backwards compatibility problem with switching it, since fd class modifications are recorded in the trace.